### ISSUE TITLE
`overfit_batches` does not work with `WeightedRandomSampler`

### ISSUE THREAD

**Issue by JackCaster** [PyTorchLightning/pytorch-lightning#10457]
## 🐛 Bug

When the sampler `WeightedRandomSampler` is used in the dataloader, the option `overfit_batches` does not work properly as a new random batch will be loaded every time.

### To Reproduce

`WeightedRandomSampler` uses a multinomial distribution under the hood to sample data points given some weights. In the command line, if I set the seed, then I am able to sample the same points

```
>>> import pytorch_lightning as pl
>>> import torch
>>> pl.seed_everything(12); torch.multinomial(torch.tensor([0.1, 1.0, 1.0, 0.1, 2.0]), 3, False)
# tensor([2, 4, 1])
>>> pl.seed_everything(12); torch.multinomial(torch.tensor([0.1, 1.0, 1.0, 0.1, 2.0]), 3, False)
# tensor([2, 4, 1])
```
Here is the boring model https://colab.research.google.com/drive/1Q64RMumll0y0hQIFDmDeW8RaPs9_ryYD?usp=sharing
Notice that I print the batch every step, and this batch is always different:

```
0 tensor([[ 5.,  5.,  5.,  5.,  5.],
        [22., 22., 22., 22., 22.],
        [35., 35., 35., 35., 35.]])

0 tensor([[39., 39., 39., 39., 39.],
        [10., 10., 10., 10., 10.],
        [ 5.,  5.,  5.,  5.,  5.]])

0 tensor([[14., 14., 14., 14., 14.],
        [15., 15., 15., 15., 15.],
        [16., 16., 16., 16., 16.]])

0 tensor([[23., 23., 23., 23., 23.],
        [15., 15., 15., 15., 15.],
        [27., 27., 27., 27., 27.]])

0 tensor([[47., 47., 47., 47., 47.],
        [20., 20., 20., 20., 20.],
        [ 2.,  2.,  2.,  2.,  2.]])
```

### Expected behavior

The same batch should be used when over fitting.

### Environment

```
* CUDA:
	- GPU:
		- Tesla T4
	- available:         True
	- version:           10.1
* Packages:
	- numpy:             1.18.5
	- pyTorch_debug:     False
	- pyTorch_version:   1.6.0+cu101
	- pytorch-lightning: 0.10.0
	- tqdm:              4.41.1
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.6.9
	- version:           #1 SMP Thu Jul 23 08:00:38 PDT 2020
```

**Comment by tchaton** [comment]
Dear @JackCaster ,

Would you mind making your notebook public ?

Best,
T.C

**Comment by JackCaster** [comment]
Done. It should work now

**Comment by rohitgr7** [comment]
@JackCaster thanks for share the example.. looks like a bug.. will investigate more :)
